devdoc/channel_internal.md

```diff
@@ -0,0 +1,196 @@
+# `channel_internal` requirements
+
+## Overview
+
+`channel_internal` is a module that contains the implementation for the `channel` module.
+
+## Exposed API
+`channel_common.h`:
+```c
+#define CHANNEL_RESULT_VALUES \
+    CHANNEL_RESULT_OK, \
+    CHANNEL_RESULT_INVALID_ARGS, \
+    CHANNEL_RESULT_ERROR
+
+MU_DEFINE_ENUM(CHANNEL_RESULT, CHANNEL_RESULT_VALUES);
+
+#define CHANNEL_CALLBACK_RESULT_VALUES \
+    CHANNEL_CALLBACK_RESULT_OK, \
+    CHANNEL_CALLBACK_RESULT_CANCELLED, \
+    CHANNEL_CALLBACK_RESULT_ABANDONED
+
+MU_DEFINE_ENUM(CHANNEL_CALLBACK_RESULT, CHANNEL_CALLBACK_RESULT_VALUES);
+
+typedef void(*PULL_CALLBACK)(void* pull_context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_PTR) data);
+typedef void(*PUSH_CALLBACK)(void* push_context, CHANNEL_CALLBACK_RESULT result);
+```
```

Comment:
Maybe it would be useful to have a line saying that the header of channel is split between a channel_common and channel internal.

Reply:
Done